Server/src/usercoordinator.erl

```diff
@@ -36,7 +47,13 @@ loop(Socket, Diagrams) ->
 	  Format_result = io_lib:format("~p", [{Did, simulation_finished, Message_number}]),
       %Sends it to the client
 	  gen_tcp:send(Socket, [Format_result ++ "~"]),
-	  loop(Socket, Diagrams)
+	  loop(Socket, Diagrams);
+	  
+	% If the user coordinator process gets an exit
+ 	% message, it either terminates, or restarts
+	% itself based on the exit reason.
+	{'EXIT', _From, normal} -> ok;
+    {'EXIT', _P, _Reason} -> loop(Socket, Diagrams) 
```

Comment:
You need to remove the crashed diagramcoordinator from the Diagrams list and then readd it